tests/CoamClient.tests.js

```diff
@@ -518,15 +518,16 @@ describe('CoamClient', function() {
 
     it('getUserPermissionsForResourceType', async function() {
         const client = new CoamClient({accessToken: accessToken});
-
-        await client.getUserPermissionsForResourceType(principal, resourceType);
+        const permissionFilter = 'test_permission';
```

Comment:
It'd be good to have 2-3 test cases since those are optional parameters. Like, is the URL correct if the 2 optional parameters are `undefined`?

Reply:
Can I have multiple it here for the same function Bcs there is a check on the top that tests that we have called the function only nce

Reply:
Sigh... that other test then needs some tweaks before you can move forward.